### PR TITLE
Add new test for Schedule:File ctor with path

### PR DESCRIPTION
**Pull request by joseph-robertson**
Pull request overview
---------------------

Add `schedule_file_2.rb`.

Link to relevant GitHub Issue(s) if appropriate:

Link to the **Ubuntu 20.04 .deb** installer to use for CI Testing. If not set, it will default to latest official release.
[OpenStudio Installer]: http://openstudio-ci-builds.s3-website-us-west-2.amazonaws.com/PR-4857/OpenStudio-3.6.0-alpha%2Bc768d9fddd-Ubuntu-20.04-x86_64.deb

This Pull Request is concerning:

 - [x] **Case 1 - `NewTest`:** a new test for a new model API class,

----------------------------------------------------------------------------------------------------------

### Case 1: New test for a new model API class

Please include which class(es) you are adding a test to specifically test for.
Include a link to the OpenStudio Pull Request in which you are adding the new classes, or the class itself if already on develop.

#### Work Checklist

The following has been checked to ensure compliance with the guidelines:

 - [ ] Tests pass either:
     - [ ] with official OpenStudio release (include version):
         - [ ] A matching OSM test has been added from the successful run of the Ruby one with the official OpenStudio release
         - [ ] The label `AddedOSM` has been added to this PR
         - [ ] All new `out.osw` have been committed

     - [ ] with current develop (incude SHA):
         - [ ] The label `PendingOSM` has been added to this PR
         - [ ] A matching OSM test has not yet been added because the official release is pending, but `model_tests.rb` has a TODO.
            ```ruby
            def test_airterminal_cooledbeam_rb
              result = sim_test('airterminal_cooledbeam.rb')
            end

            # TODO: To be added in the next official release after: 2.5.0
            # def test_airterminal_fourpipebeam_osm
            #   result = sim_test('airterminal_fourpipebeam.osm')
            # end
            ```
        - [ ] No `out.osw` have been committed as they need to be run with an official OpenStudio version

 - [ ] **Ruby test is stable**: when run multiple times on the same machine, it produces the same total site kBTU.
    Please paste the heatmap png generated after running the following commands:
     - [ ] I ensured that I assign systems/loads/etc in a repeatable manner (eg: if I assign stuff to thermalZones, I do `model.getThermalZones.sort_by{|z| z.name.to_s}.each do ...` so I am sure I put the same ZoneHVAC systems to the same zones regardless of their order)
     - [ ] I tested stability using `process_results.py` (see `python process_results.py --help` for usage).
    Please paste the text output or heatmap png generated after running the following commands:
        ```bash
        # Clean up all custom-tagged OSWs
        python process_results.py test-stability clean
        # Run your test 5 times in a row. Replace `testname_rb` (eg `airterminal_fourpipebeam_rb`)
        python process_results.py test-stability run -n testname_rb
        # Check that they all passed
        python process_results.py test-status --tagged
        # Check site kBTU differences
        python process_results.py heatmap --tagged

        ```

 - [ ] **Object has been added to `autosize_hvac.rb` to ensure the autosizedXXX values methods do work**

----------------------------------------------------------------------------------------------------------
### Review Checklist

 - [ ] Code style (indentation, variable names, strip trailing spaces)
 - [ ] Functional code review (it has to work!)
 - [ ] Matching OSM test has been added or `# TODO` added to `model_tests.rb`
 - [ ] Appropriate `out.osw` have been committed
 - [ ] Test is stable
 - [ ] Object is tested in `autosize_hvac` as appropriate
 - [ ] The appropriate labels have been added to this PR:
   - [ ] One of: `NewTest`, `TestFix`, `NewTestForExisting`, `Other`
   - [ ] If `NewTest`: add `PendingOSM` or `AddedOSM`

### CODE REVIEW

File: model/simulationtests/schedule_file_2.rb [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+model = BaselineModel.new
+
+# make a 2 story, 100m X 50m, 10 zone core/perimeter building
+model.add_geometry({ 'length' => 100,
+                     'width' => 50,
+                     'num_floors' => 2,
+                     'floor_to_floor_height' => 4,
+                     'plenum_height' => 1,
+                     'perimeter_zone_depth' => 3 })
```
Comment (jmarrec) [inline model/simulationtests/schedule_file_2.rb]: Make it a one zone please.

Comment (jmarrec) [reply model/simulationtests/schedule_file_2.rb]: Unless we don't plan on merging this one actually. Maybe we don't need to? I think we should probably keep it.

File: model/simulationtests/schedule_file_2.rb [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+                    'offset' => 1,
+                    'application_type' => 'Above Floor' })
+
+# add ASHRAE System type 03, PSZ-AC
+model.add_hvac({ 'ashrae_sys_num' => '03' })
+
+# add thermostats
+model.add_thermostats({ 'heating_setpoint' => 24,
+                        'cooling_setpoint' => 28 })
+
```
Comment (jmarrec) [inline model/simulationtests/schedule_file_2.rb]: ideal air loads (or nothing) would probably do. See adiabatic_construction_set.rb for a minimal one

File: model_tests.rb [hunk 3]
```diff
@@ -1032,6 +1032,15 @@ def test_schedule_file_osm
...
   end
 
+  def test_schedule_file_2_rb
+    result = sim_test('schedule_file_2.rb')
+  end
+
+  # TODO: To be added in the next official release after: 3.5.1
+  # def test_schedule_file_2_osm
+  # result = sim_test('schedule_file_2.osm')
+  # end
```
Comment (jmarrec) [inline model_tests.rb:1042]: :+1:

**Review by jmarrec — CHANGES_REQUESTED** [review]

File: model/simulationtests/schedule_file_2.rb [hunk 4]
```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+require 'openstudio'
```
Comment (jmarrec) [inline model/simulationtests/schedule_file_2.rb:5]:
```suggestion

# This test aims to test the new ScheduleFile ctor that takes a direct path and does not make a copy to the companion directory. Cf NREL/OpenStudio#4857

require 'openstudio'
```

**Review by jmarrec — CHANGES_REQUESTED** [review]

File: model/simulationtests/schedule_file_2.rb [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+###############################################################################
+
+# set year to 2013
+yd = model.getYearDescription
+yd.setCalendarYear(2013)
+
+# add schedule file
+file_name = File.join(File.dirname(__FILE__), 'lib/schedulefile.csv')
+file_name = File.realpath(file_name)
+schedule_file = OpenStudio::Model::ScheduleFile.new(model, file_name, 3, 1)
```
Comment (joseph-robertson) [inline model/simulationtests/schedule_file_2.rb:58]: Why doesn't this line fail? The new ScheduleFIle ctor takes an openstudio path, not a string...

Comment (jmarrec) [reply model/simulationtests/schedule_file_2.rb:58]: (C++) Implicit conversion from std::string to path. See MCVE https://gcc.godbolt.org/z/79qYxd5e4

**Comment by jmarrec** [comment]
I can't recall if we want to merge this one or if it was a one-off test to make sure a change was ok. Do we just close or do we upgrade and merge?

**Comment by joseph-robertson** [comment]
Probably safe to just close, as this is not really a regression type test.

**Pull request closed without merging**